src/hotspot/share/oops/constantPool.cpp

```diff
@@ -1286,6 +1286,13 @@ bool ConstantPool::compare_entry_to(int index1, const constantPoolHandle& cp2,
   jbyte t1 = tag_at(index1).non_error_value();
   jbyte t2 = cp2->tag_at(index2).non_error_value();
 
+  if (t1 == JVM_CONSTANT_Class) {
+    // Some classes are pre-resolved (like Throwable) which may lead to
+    // consider it as a different entry. we then revert them back temporarily
+    // to ensure proper comparison.
+    t1 = JVM_CONSTANT_UnresolvedClass;
+  }
+
```

Comment:
All consequences of this change are not clear to me yet.
The lines 1307-1314 become not needed anymore.
Also, should the same be done for t2 as well?

Reply:
t2 could be a resolved class, and can be compared with unresolved class because the function klass_name_at() works for both.  It might be a good idea to change both of them though, although not necessary imo.
You're right 1307-1314 are not reached anymore.  Neither is the ClassIndex case but not to remove as part of this change.